iceoryx_posh/source/popo/building_blocks/condition_variable_waiter.cpp

```diff
@@ -27,21 +27,40 @@ ConditionVariableWaiter::ConditionVariableWaiter(cxx::not_null<ConditionVariable
 void ConditionVariableWaiter::reset() noexcept
 {
     // Count the semaphore down to zero
-    while (getMembers()->m_semaphore.tryWait())
+    while (getMembers()
+               ->m_semaphore.tryWait()
+               .or_else([](posix::SemaphoreError) {
+                   errorHandler(Error::kPOPO__CONDITION_VARIABLE_WAITER_SEMAPHORE_CORRUPTED_IN_RESET,
+                                nullptr,
+                                ErrorLevel::FATAL);
+               })
+               .get_value())
     {
     }
 }
 
 void ConditionVariableWaiter::wait() noexcept
 {
-    getMembers()->m_semaphore.wait();
+    if (getMembers()->m_semaphore.wait().has_error())
+    {
+        errorHandler(Error::kPOPO__CONDITION_VARIABLE_WAITER_SEMAPHORE_CORRUPTED_IN_WAIT, nullptr, ErrorLevel::FATAL);
+    }
 }
 
 bool ConditionVariableWaiter::timedWait(units::Duration timeToWait) noexcept
 {
     auto timeout = timeToWait.timespec(units::TimeSpecReference::Epoch);
     auto continueOnInterrupt{false};
-    return getMembers()->m_semaphore.timedWait(&timeout, continueOnInterrupt);
+    auto result = getMembers()->m_semaphore.timedWait(&timeout, continueOnInterrupt);
```

Comment:
so we can have spurious wakeups?

Reply:
Spurious wakeups are handled in `timedWait` inside the `semaphore` and cannot occur on this level.